core/domain/exp_services.py

```diff
@@ -316,6 +317,52 @@ def get_multiple_explorations_by_id(exp_ids, strict=True):
     return result
 
 
+def get_exploration_and_exploration_rights_by_id(exploration_id, version=None):
+    """Returns a tuple for exploration domain object and exploration rights
+        object.
+
+    Args:
+        exploration_id: str. Id of the exploration.
+        version: str. Version of the exploration.
+
+    Returns:
+        Tuple(exploration object or None, exploration rights object or None).
```

Comment:
Same comments as before (throughout the entire function).

Reply:
done.